src/ethereum/frontier/vm/gas.py

```diff
@@ -118,3 +124,50 @@ def calculate_gas_extend_memory(
     total_cost = calculate_memory_gas_cost(after_size)
     to_be_paid = total_cost - already_paid
     return to_be_paid
+
+
+def calculate_call_gas_cost(
+    state: State, gas: U256, to: Address, value: U256
+) -> U256:
+    """
+    Calculates the gas amount for executing Opcodes `CALL` and `CALLCODE`.
+
+    Parameters
+    ----------
+    state :
+        The current state.
+    gas :
+        The amount of gas provided to the message-call.
+    to:
+        The address of the recipient account.
+    value:
+        The amount of `ETH` that needs to be transferred.
+
+    Returns
+    -------
+    call_gas_cost: `ethereum.base_types.U256`
+        The total gas amount for executing Opcodes `CALL` and `CALLCODE`.
+    """
+    _account_exists = account_exists(state, to)
+    create_gas_cost = GAS_NEW_ACCOUNT if not _account_exists else U256(0)
+    transfer_gas_cost = GAS_CALL_VALUE if value != 0 else U256(0)
+    return GAS_CALL + gas + create_gas_cost + transfer_gas_cost
+
+
+def calculate_message_call_gas_cost(gas: U256, value: U256) -> U256:
+    """
+    Calculates the gas amount for making the message call.
+
+    Parameters
+    ----------
+    gas :
+        The amount of gas provided to the message call.
+    value:
+        The amount of `ETH` that needs to be transferred.
+
+    Returns
+    -------
+    message_call_gas_cost: `ethereum.base_types.U256`
+        The total gas amount for making the message-call.
+    """
+    return gas + (GAS_CALL_STIPEND if value != 0 else U256(0))
```

Comment:
Could we do something like:

```suggestion
def calculate_message_call_gas_stipend(value: U256) -> U256:
    """
    Calculates the gas stipend for making the message call with the given value.

    Parameters
    ----------
    value:
        The amount of `ETH` that needs to be transferred.

    Returns
    -------
    message_call_gas_stipend: `ethereum.base_types.U256`
        The gas stipend for making the message-call.
    """
    return U256(0) if value == 0 else GAS_CALL_STIPEND
```

Would need to do the addition in the caller though.